lib/rubygems.rb

```diff
@@ -773,6 +773,10 @@ def self.write_binary(path, data)
     open_file(path, 'wb') do |io|
       io.write data
     end
+  rescue Errno::ENOSPC => e
+    # If the write failed, delete the file -- it's *guaranteed* to be corrupted.
+    File.delete(path) if File.exist?(path)
+    raise e
```

Comment:
When raising the same error rescued, I generally prefer to let the original exception flow, rather than raising a duplicated one.

```suggestion
    raise
```

Reply:
Sounds good to me. I'll push a change removing the ` => e` and changing `raise e` to `raise`. 👍